WordPress/Classes/ViewRelated/NUX/SignupEpilogueViewController.swift

```diff
@@ -0,0 +1,127 @@
+import UIKit
+
+class SignupEpilogueViewController: NUXViewController {
+
+    // MARK: - Properties
+
+    private var buttonViewController: NUXButtonViewController?
+    private var updatedDisplayName: String?
+    private var updatedPassword: String?
+
+    // MARK: - View
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: false)
+    }
+
+    // MARK: - Navigation
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        super.prepare(for: segue, sender: sender)
+
+        if let vc = segue.destination as? NUXButtonViewController {
+            buttonViewController = vc
+            buttonViewController?.delegate = self
+            buttonViewController?.setButtonTitles(primary: NSLocalizedString("Continue", comment: "Button text on site creation epilogue page to proceed to My Sites."))
+        }
+
+        if let vc = segue.destination as? SignupEpilogueTableViewController {
+            vc.loginFields = loginFields
+            vc.delegate = self
+        }
+    }
+
+}
+
+// MARK: - NUXButtonViewControllerDelegate
+
+extension SignupEpilogueViewController: NUXButtonViewControllerDelegate {
+    func primaryButtonPressed() {
+
+        if updatedDisplayName != nil || updatedPassword != nil {
+            updateUserInfo()
+        } else {
+            self.navigationController?.dismiss(animated: true, completion: nil)
+        }
+    }
+}
+
+// MARK: - SignupEpilogueTableViewControllerDelegate
+
+extension SignupEpilogueViewController: SignupEpilogueTableViewControllerDelegate {
+    func displayNameUpdated(newDisplayName: String) {
+        updatedDisplayName = newDisplayName
+    }
+
+    func passwordUpdated(newPassword: String) {
+        updatedPassword = newPassword
+    }
+
+}
+
+// MARK: - Private Extension
+
+private extension SignupEpilogueViewController {
+
+    func updateUserInfo() {
+
+        guard let updatedDisplayName = updatedDisplayName else {
```

Comment:
This method is called when `updatedDisplayName != nil || updatedPassword != nil` which means it has to work if `updatedDisplayName` is nil. 

I triggered this by editing only the password and not the display name, and the continue button does nothing.